addon/components/polaris-data-table.js

```diff
@@ -0,0 +1,535 @@
+   * @default false
+   * @private
+   */
+  collapsed: false,
+
+  /**
+   * @property columnVisibilityData
+   * @type {Object[]}
+   * @private
+   */
+  columnVisibilityData: null,
+
+  /**
+   * @property previousColumn
+   * @type {Object}
+   * @private
+   */
+  previousColumn: null,
+
+  /**
+   * @property currentColumn
+   * @type {Object}
+   * @private
+   */
+  currentColumn: null,
+
+  /**
+   * @property sorted
+   * @type {boolean}
+   * @default false
+   * @private
+   */
+  sorted: false,
+
+  /**
+   * @property sortedColumnIndex
+   * @type {Number}
+   * @private
+   */
+  sortedColumnIndex: null,
+
+  /**
+   * @property sortDirection
+   * @type {String}
+   * @private
+   */
+  sortDirection: null,
+
+  /**
+   * @property heights
+   * @type {Number[]}
+   * @private
+   */
+  heights: null,
+
+  /**
+   * @property preservedScrollPosition
+   * @type {Object}
+   * @private
+   */
+  preservedScrollPosition: null,
+
+  /**
+   * @property previousTruncate
+   * @type {boolean}
+   * @private
+   */
+  previousTruncate: null,
+
+  /**
+   * @property totalsRowHeading
+   * @type {String}
+   * @private
+   */
+  totalsRowHeading: 'Totals',
+
+  /**
+   * @property eventHandlerParams
+   * @type {Array[]}
+   * @private
+   */
+  eventHandlerParams,
+
+  /**
+   * @property dataTable
+   * @type {HTMLElement}
+   * @private
+   */
+  dataTable: volatileElementLookup('.Polaris-DataTable').readOnly(),
+
+  /**
+   * @property scrollContainer
+   * @type {HTMLElement}
+   * @private
+   */
+  scrollContainer: volatileElementLookup('.Polaris-DataTable__ScrollContainer').readOnly(),
+
+  /**
+   * @property table
+   * @type {HTMLElement}
+   * @private
+   */
+  table: volatileElementLookup('.Polaris-DataTable__Table').readOnly(),
+
+  /**
+   * @property contentTypes
+   * @type {String[]}
+   * @private
+   */
+  contentTypes: computed('columnContentTypes.[]', function() {
+    let columnContentTypes = this.get('columnContentTypes');
+    let fixedCellType = columnContentTypes[0];
+
+    return [ fixedCellType, ...columnContentTypes ];
+  }).readOnly(),
+
+  /**
+   * @property scrollContainerStyle
+   * @type {String}
+   * @private
+   */
+  scrollContainerStyle: computed('footerContent', 'heights.[]', function() {
+    if (isBlank(this.get('footerContent'))) {
+      return null;
+    }
+
+    return htmlSafe(`margin-bottom: ${ this.get('heights.lastObject') }px;`);
+  }).readOnly(),
+
+  resetScrollPosition() {
+    let { left, top } = this.get('preservedScrollPosition');
+
+    if (left) {
+      this.get('scrollContainer').scrollLeft = left;
+    }
+
+    if (top) {
+      window.scrollTo(0, top);
+    }
+  },
+
+  setHeightsAndScrollPosition() {
+    this.set('heights', this.tallestCellHeights());
+
+    scheduleOnce('afterRender', this, this.resetScrollPosition);
+  },
+
+  calculateColumnVisibilityData(collapsed) {
+    if (collapsed) {
+      let headerCells = this.get('table').querySelectorAll(
```

Comment:
we can make this a one-liner?

Reply:
This is just a tweak from what's in the [React source](https://github.com/Shopify/polaris/blob/master/src/components/DataTable/DataTable.tsx#L337), I wanted to keep the two implementations similar to make changes more visible in the future...

Reply:
well it's still the same thing just one-line no?

Reply:
Yeah, I literally just copied the React code and changed the bits that needed it. Can go through and tweak the styling of all the copied code if you want (will fix the "spaces" nit comment on line 51 as well)

Reply:
however you feel too...we wont' be able to compare code through a git-diff style anyway....so I'd be for these small tweaks, but flexible